### PR TITLE
Add state flow charts to Gesture Handler's documentation.

### PR DESCRIPTION
**Pull request by LatekVo**
## Description

Add a flow chart to visualise state changes of gesture components.

## Test plan

Run docs locally, go to the `Gesture states & events` article.

### compatible with both Light and Dark modes

https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/6c4794ac-3f02-4990-9d42-e8602c6cb10a

### adaptive layout

https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/3294d39a-0357-41c4-867c-39b9c3d89781

### CODE REVIEW

**Comment by LatekVo** [comment]
# This is how this project currently looks like:
https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/1e4bff1e-773e-44be-acc9-a4b5f5ca94d6

**Comment by LatekVo** [comment]
Found some critical errors, please give me some more time before reviewing :/

**Comment by LatekVo** [comment]
we found some bugs with @m-bert which likely are a problem within the handlers themselves.
Under some circumstances, a direct transition from `BEGAN` to `CANCELLED` is possible, which doesn't follow GH's documentation. Same happens with `ACTIVE -> FAILED`, which shouldn't be possible at all.

**Comment by LatekVo** [comment]
there is currently an issue with svg antialiasing which causes those 22deg lines to be much thicker than both 45 and 0 deg lines. I resolved this by turning off AA and applying a 0.2px blur to the affected elements, but this is a very unclean solution.
Please tell me if you know a better way, but i found none.

File: docs/src/examples/ChartElement.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  chartManager,
+  primaryColor,
+  highlightColor,
+  innerRef,
+  style,
+}: ChartElementProps) {
+  const progress = useSharedValue(0);
+
+  useEffect(() => {
+    // this listener is cleared through FlowChart
```
Comment (m-bert) [inline docs/src/examples/ChartElement.tsx:32]:
```suggestion
```

File: docs/src/examples/ChartManager.ts [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    handle.elementIds.set(State.CANCELLED, cancelledId);
+    handle.elementIds.set(State.UNDETERMINED, undeterminedId);
+
+    undeterminedCallback(true);
+
+    const WAVE_DELAY_MS = 100;
+
+    const resetAllStates = (event: GestureStateChangeEvent<any>) => {
+      undeterminedCallback(true);
+      if (event.state == 1) {
```
Comment (m-bert) [inline docs/src/examples/ChartManager.ts]:
```suggestion
      if (event.state == State.FAILED) {
```
We should use `State` enum.

**Comment by LatekVo** [comment]
### Does this style look better?

https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/19becf01-9860-4bab-805b-71399af9bcde

**Comment by LatekVo** [comment]
https://github.com/software-mansion/react-native-gesture-handler/assets/74246391/214b1f48-6588-44d4-983a-37c7100a5e09

**Review by patrycjakalinska — APPROVED** [review]
My suggested changes are more of a stylistic/user experience nature, but state flow is working and looks amazing 🚀

File: docs/src/components/CollapsibleCode/index.tsx
[the remark this answers is not in the document]
Comment (kacperkapusciak) [reply docs/src/components/CollapsibleCode/index.tsx:1]: if the code is fully shown then it doesn't make sense to use the CollapsibleCode component at all ;)

File: docs/src/examples/FlowChart.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+              key={connection.id}
+              startPoint={{
+                x: itemsCoordsRef.current[connection.from].x,
+                y: itemsCoordsRef.current[connection.from].y,
+              }}
+              endPoint={{
+                x: itemsCoordsRef.current[connection.to].x,
+                y: itemsCoordsRef.current[connection.to].y,
+              }}
+            />
```
[the remark this answers is not in the document]
Comment (kacperkapusciak) [reply docs/src/examples/FlowChart.tsx]: fixed in [d5649d4](https://github.com/software-mansion/react-native-gesture-handler/pull/2817/commits/d5649d4456f2b94b075652714b0ab9b1014131a8)

**Review by kacperkapusciak — APPROVED** [review]
the code is overly complex, highly unmaintainable, and made completely not in a "react way"

lgtm, ship it ✅

**Pull request merged**